.ci/Jenkinsfile

```diff
@@ -115,28 +122,31 @@ pipeline {
   }
   post {
     cleanup {
-      notifyBuildResult()
+      notifyBuildResult(downstreamJobs: itsDownstreamJobs)
     }
   }
 }
 
 def runJob(agentName, buildOpts = ''){
-  def branch = env.BRANCH_NAME
-
-  if (env.CHANGE_ID) {
-    branch = env.CHANGE_TARGET
+  def branch = env.CHANGE_ID?.trim() ? env.CHANGE_TARGET : env.BRANCH_NAME
+  def job
+  try {
```

Comment:
this block is repeatable on each job we check for build cause maybe we need a step to encapsulate it

```
catchDownstreamErrors('Downstream job name', registryVar) {
...
}
```

Reply:
Not sure If I understand it correctly.

Something like the below snippet?

```
def runJob(agentName, buildOpts = ''){
   catchDownstreamErrors("Downstream job ${agentName}", itsDownstreamJobs) {
        def branch = env.CHANGE_ID?.trim() ? env.CHANGE_TARGET : env.BRANCH_NAME
        def quiet = randomNumber(min: 2, max: 15) + 5
        job = build(job: "apm-integration-test-downstream/${branch}",
        parameters: [
        string(name: 'AGENT_INTEGRATION_TEST', value: agentName),
        string(name: 'ELASTIC_STACK_VERSION', value: params.ELASTIC_STACK_VERSION),
        string(name: 'INTEGRATION_TESTING_VERSION', value: "${env.GIT_BASE_COMMIT}"),
        string(name: 'MERGE_TARGET', value: "${branch}"),
        string(name: 'BUILD_OPTS', value: "${params.BUILD_OPTS} ${buildOpts}"),
        string(name: 'UPSTREAM_BUILD', value: currentBuild.fullDisplayName),
        booleanParam(name: 'DISABLE_BUILD_PARALLEL', value: '')],
        propagate: true,
        quietPeriod: quiet,
        wait: true)
    }
}


// Step catchDownstreamErrors

call( ... ) {
   try {
       body()
    } catch(e) {

   } finally {
       // add the buildInfo details
   }
```

If so, I'm not sure if we can get the buildInfo details easily

Reply:
yep that exactly, we can access to the data we can on the pipeline